app/components/post_list/post/body/content/permalink_preview/permalink_preview.tsx

```diff
@@ -0,0 +1,197 @@
+
+import React, {useMemo} from 'react';
+import {Text, View, Pressable} from 'react-native';
+
+import EditedIndicator from '@components/EditedIndicator';
+import FormattedTime from '@components/formatted_time';
+import ProfilePicture from '@components/profile_picture';
+import {useTheme} from '@context/theme';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+import {displayUsername} from '@utils/user';
+
+import type UserModel from '@typings/database/models/servers/user';
+
+type PermalinkPreviewProps = {
+    embedData: PermalinkEmbedData;
+    showPermalinkPreviews: boolean;
+    author?: UserModel;
+    locale: string;
+    teammateNameDisplay: string;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            backgroundColor: theme.centerChannelBg,
+            borderColor: changeOpacity(theme.centerChannelColor, 0.2),
+            borderWidth: 1,
+            borderRadius: 8,
+            marginTop: 8,
+            marginBottom: 8,
+            padding: 12,
+        },
+        header: {
+            flexDirection: 'row',
+            alignItems: 'center',
+            marginBottom: 8,
+        },
+        authorInfo: {
+            flex: 1,
+            marginLeft: 12,
+            flexDirection: 'row',
+            alignItems: 'center',
+        },
+        authorName: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100, 'SemiBold'),
+        },
+        timestamp: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75),
+            marginLeft: 8,
+        },
+        messageContainer: {
+            marginBottom: 8,
+        },
+        messageText: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100),
+            lineHeight: 20,
+        },
+
+        channelContext: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75),
+        },
+        channelName: {
+            color: theme.linkColor,
+            ...typography('Body', 75, 'SemiBold'),
+        },
+    };
+});
+
+const PermalinkPreview = ({embedData, showPermalinkPreviews, author, locale, teammateNameDisplay}: PermalinkPreviewProps) => {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    if (!showPermalinkPreviews) {
+        return null;
+    }
+
+    const {
+        post,
+        channel_display_name,
+        channel_type,
+    } = embedData;
+
+    const {formattedMessage} = useMemo(() => {
+        const message = post?.message;
+
+        if (!message || typeof message !== 'string') {
+            return {formattedMessage: ''};
+        }
+
+        const cleanMessage = message.trim();
+
+        const lines = cleanMessage.split('\n');
+        if (lines.length > 4) {
```

Comment:
Can you explain this magic number of 4?  Or better yet, make a const out of it with the proper variable name so that user in the future would know what the 4 is for.

Reply:
Added const `MAX_PERMALINK_PREVIEW_LINES`.